kvbc/cmf/categorized_kvbc_msgs.cmf

```diff
@@ -22,39 +22,88 @@ Msg SharedValueData 4{
     list string categories_ids
 }
 
-Msg SharedUpdatesData 5{
+Msg SharedKeyValueUpdatesData 5{
     map string SharedValueData kv
     bool calculate_hash
 }
 
 
 # Updates info
 
-Msg MerkleKeyFlag 6{
+Msg MerkleKeyFlag 10{
     bool deleted
 }
 
-Msg MerkleUpdatesInfo 7{
+Msg MerkleUpdatesInfo 11{
     map string MerkleKeyFlag keys
-    fixedlist uint8 32 hash
-    uint64 version
+    fixedlist uint8 32 state_hash
+    uint64 state_root_version
 }
 
-Msg KVKeyFlag 8{
+Msg KVKeyFlag 12{
     bool deleted
     bool stale_on_update
 }
 
-Msg KeyValueUpdatesInfo 9{
+Msg KeyValueUpdatesInfo 13{
     map string KVKeyFlag keys
     optional fixedlist uint8 32 hash
 }
 
-Msg SharedKeyData 10 {
+Msg SharedKeyData 14 {
     list string categories
 }
 
-Msg SharedKeyValueUpdatesInfo 11{
+Msg SharedKeyValueUpdatesInfo 15{
     map string SharedKeyData keys
     optional fixedlist uint8 32 hash
+}
+
+# Blocks
+
+Msg BlockKey 20 {
+    uint64 block_id
+}
+
+Msg BlockData 21{
+    uint64 block_id
+    fixedlist uint8 32 parent_digest
+    optional SharedKeyValueUpdatesInfo shared_updates_info
+    map string oneof{
+        MerkleUpdatesInfo
+        KeyValueUpdatesInfo
+    } categories_updates_info
+}
+
+Msg RawBlockMerkleUpdates 22 {
+    MerkleUpdatesData updates
+    fixedlist uint8 32 state_hash
```

Comment:
Nit: Maybe we can call all hashes for all categories `root_hash`.

Reply:
yep